#!/usr/bin/env python
PACKAGE = "reconfigure_picamera"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("camera_framerate", int_t, 0, "", 60,  1, 100)
gen.add("camera_index",     int_t, 0, "", 0,  0, 8)
gen.add("camera_topic",     str_t,    0, "Camera topic to publish",  "/pi_camera_0")
gen.add("publish_image",    bool_t,    0, "Start publishing raw images",  False)

gen.add("hsv_threshold_lower_hue",          int_t, 0, "", 0,  0, 180)
gen.add("hsv_threshold_lower_saturation",   int_t, 0, "", 115,  0, 255)
gen.add("hsv_threshold_lower_value",        int_t, 0, "", 0,  0, 255)

gen.add("hsv_threshold_upper_hue",          int_t, 0, "", 10,  0, 180)
gen.add("hsv_threshold_upper_saturation",   int_t, 0, "", 255,  0, 255)
gen.add("hsv_threshold_upper_value",        int_t, 0, "", 255,  0, 255)

width_enum = gen.enum([  gen.const("Tiny",       int_t, 320,  "Tiny Resolution 320x240"),
                         gen.const("Small",      int_t, 640,  "Small Resolution 640x480"),
                         gen.const("Medium",     int_t, 1024, "Medium Resolution 1024x768"),
                         gen.const("Large",      int_t, 1280, "Large Resolution 1280x720"),
                         gen.const("ExtraLarge", int_t, 1920, "ExtraLarge Resolution 1920x1080")],
                     "An enum to set camera resolution")

height_enum = gen.enum([ gen.const("Tiny",       int_t, 240,  "Tiny Resolution 320x240"),
                         gen.const("Small",      int_t, 480,  "Small Resolution 640x480"),
                         gen.const("Medium",     int_t, 768, "Medium Resolution 1024x768"),
                         gen.const("Large",      int_t, 720, "Large Resolution 1280x720"),
                         gen.const("ExtraLarge", int_t, 1080, "ExtraLarge Resolution 1920x1080")],
                     "An enum to set camera resolution")

gen.add("camera_resolution_width",  int_t, 0, "Camera Resolution based on enum: Tiny, Small, Medium, Large, XLarge", 640, 320, 1920, edit_method=width_enum)
gen.add("camera_resolution_height", int_t, 0, "Camera Resolution based on enum: Tiny, Small, Medium, Large, XLarge", 480, 240, 1080, edit_method=height_enum)

exit(gen.generate(PACKAGE, "harvey_move_to_see", "ReconfigurePicamera"))
